tests/integration/_experimental/object_detection/test_dataset.py

```diff
@@ -40,51 +119,41 @@ def _assert_result_bbox_contains_fields(df_results: pd.DataFrame, columns: List[
 
 
 @pytest.mark.metrics
-def test__upload_results__single_class() -> None:
-    name = with_test_prefix(f"{__file__}::test__upload_results__single_class")
+@pytest.mark.parametrize(
+    "annotation,gts,infs",
+    [
+        ("labeled_bboxes", gt_labeled_bbox_single, inf_labeled_bbox_single),
+        ("unlabeled_bboxes", gt_unlabeled_bbox_single, inf_unlabeled_bbox_single),
+        ("labeled_polygons", gt_labeled_polygon_single, inf_labeled_polygon_single),
+        ("unlabeled_polygons", gt_unlabeled_polygon_single, inf_unlabeled_polygon_single),
+    ],
+)
+def test__upload_results__single_class(annotation, gts, infs) -> None:
```

Comment:
nit: missing type hints for the inputs, seems like we have some unintentional mypy checking skip